tests/admin_views/tests.py

```diff
@@ -188,11 +188,13 @@ def assertContentBefore(self, response, text1, text2, failing_msg=None):
 class AdminViewBasicTest(AdminViewBasicTestCase):
     def test_trailing_slash_required(self):
         """
-        If you leave off the trailing slash, app should redirect and add it.
+        If you leave off the trailing slash, app will not redirect as that could be
+        used to enumerate models that the user does not have permission to know
+        about.
         """
         add_url = reverse('admin:admin_views_article_add')
         response = self.client.get(add_url[:-1])
-        self.assertRedirects(response, add_url, status_code=301)
+        self.assertEqual(response.status_code, 404)
```

Comment:
This is the unauthenticated case yes? What's the story for the authenticated user? (Ah, I see the release note...)

Reply:
This is the authenticated case. The user is logged in `AdminViewBasicTest.setUp`. I added `test_known_url_missing_slash_redirects_login_if_not_authenticated` to demonstrate the unauthenticated case. It is a redirect to the login.

Reply:
So, I would revert this.